zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -352,16 +356,18 @@ public Set<String> getAllAuthorizedResourceIdentifiers(final AuthorizationReques
    * type. This does not include inherited authorizations, for example authorizations for users from
    * assigned roles or groups.
    */
-  public Set<String> getDirectAuthorizedResourceIdentifiers(
+  public Set<AuthorizationScope> getDirectAuthorizedResourceIdentifiers(
       final AuthorizationOwnerType ownerType,
       final String ownerId,
       final AuthorizationResourceType resourceType,
       final PermissionType permissionType) {
-    return authorizationState.getResourceIdentifiers(
-        ownerType, ownerId, resourceType, permissionType);
+    return authorizationState
+        .getAuthorizationScopes(ownerType, ownerId, resourceType, permissionType)
+        .stream()
+        .collect(Collectors.toSet());
```

Comment:
🔧 The authorization state already returns a set of `AuthorizationScope`. We can just return this instead of having the stream collect here.